Day0/bipul018/assignment1/cjpt.c

```diff
@@ -0,0 +1,134 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+
+struct color{
+    unsigned char r;
+    unsigned char g;
+    unsigned char b;
+};
+int main(int argc,char *argv[])
+{
+
+    //default filenames will be used if not provided in command line
+    char source[100];
+    if(argc>1)
+    {
+        strcpy(source,argv[1]);
+    }
+    else
+    {
+        strcpy(source,"render.txt");
+    }
+    FILE *fp;
+    fp=fopen(source,"r");
```

Comment:
It is recommended to use "rb" (binary format) instead of "r". More on that here: https://stackoverflow.com/questions/229924/difference-between-files-written-in-binary-and-text-mode
Some functions such as ftell does not work well when you are using text mode, more on those here: https://en.cppreference.com/w/c/io/ftell